apps/opik-backend/src/main/java/com/comet/opik/api/Span.java

```diff
@@ -65,4 +66,17 @@ public static class Write {
         public static class Public {
         }
     }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        Span span = (Span) o;
+        return Objects.equals(id, span.id);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(id);
+    }
```

Comment:
This custom implementation can lead to a lot of confusion and problems. You should keep the default equals and hascode implementation of this record class.

If we want to dedupe or validate just based on the ID field, we should use some other implementation. Maybe a custom Java `Comparator` could be a good choice.

Reply:
Same